### PR TITLE
:recycle: refactor api client

### PR DESCRIPTION
**Pull request by chassing**
Move parts of the `GlitchtipClient` into a separate base class. The future `UnleashServer` API client will reuse those base parts.

### CODE REVIEW

File: reconcile/utils/rest_api_base.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        host: str,
+        token: str,
+        max_retries: int | None = None,
+        read_timeout: float | None = None,
+    ) -> None:
+        self.host = host
+        self.token = token
+        self.max_retries = max_retries if max_retries is not None else 3
+        self.read_timeout = read_timeout if read_timeout is not None else 30
+        self._thread_local = threading.local()
```
Comment (hemslo) [inline reconcile/utils/rest_api_base.py:34]: now we are making this as a base, let's fix the session issue to make it use the same session object and do auto close with context manager.

File: reconcile/utils/rest_api_base.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    * https://gitlab.com/glitchtip/glitchtip-backend/-/blob/master/glitchtip/pagination.py#L34
+    * https://requests.readthedocs.io/en/latest/api/?highlight=links#requests.Response.links
+    """
+    if links.get("next", {}).get("results", "false") == "true":
+        return links["next"]["url"]
+    return None
+
+
+class ApiBase:
+    """This class provides a common standard for REST API clients."""
```
Comment (hemslo) [inline reconcile/utils/rest_api_base.py:21]: Do you want to make this as the base for all api clients we use, I have a similar one can benefit from the base, https://github.com/app-sre/qontract-reconcile/blob/master/tools/cli_commands/cost_report/cost_management_api.py, but it requires oauth token handling and customized retry (backoff and status)

Comment (chassing) [reply reconcile/utils/rest_api_base.py:37]: I've added `session` and `auth` attributes to `__init__` to customize those things easily.

File: reconcile/utils/rest_api_base.py (hunk 1 above)
Comment (chassing) [reply reconcile/utils/rest_api_base.py]: Yes, you're right. Changed that in the last commit.

File: reconcile/utils/rest_api_base.py (hunk 2 above)
Comment (hemslo) [reply reconcile/utils/rest_api_base.py:37]: Thank you! I'll try to use the new base for cost  management api!

**Review by hemslo — APPROVED** [review]
LGTM!

**Pull request merged**